### PR TITLE
feat(types): Add TypeScript support for addhoc

### PR DESCRIPTION
**Pull request by shawnkoon**
Hello 👋, I really like this project and wanted to see if we could add TypeScript support to it!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,6 +38,7 @@
     "@babel/preset-env": "^7.5.4",
     "@babel/preset-react": "^7.0.0",
     "@babel/register": "^7.4.4",
+    "@types/react": "^16.9.2",
```
Comment (decompil3d) [inline package.json]: Should this match the version of react pulled in below? Obviously, npm will just grab latest satisfying semver, but it seems like it's safest for these two to match.

```suggestion
    "@types/react": "^16.8.6",
```

File: package.json [hunk 2]
```diff
@@ -6,11 +6,13 @@
...
   "module": "./index",
   "react-native": "./index",
+  "types": "./index.d.ts",
   "scripts": {
     "build": "babel -d lib index.js",
     "prepublishOnly": "npm run build",
     "lint": "eslint --fix *.js",
-    "test": "nyc mocha --require setup-env test/*.test.js",
+    "typings:test": "tsc --noEmit ./index.d.ts",
+    "test": "nyc mocha --require setup-env test/*.test.js && npm run typings:test",
```
Comment (decompil3d) [inline package.json]: Let's split out the mocha tests if we're doing this
```suggestion
    "test:mocha": "nyc mocha --require setup-env test/*.test.js",
    "test": "npm run test:mocha && npm run test:types",
```

File: package.json [hunk 3]
```diff
@@ -6,11 +6,13 @@
   "browser": "./lib/index",
   "module": "./index",
   "react-native": "./index",
+  "types": "./index.d.ts",
   "scripts": {
     "build": "babel -d lib index.js",
     "prepublishOnly": "npm run build",
     "lint": "eslint --fix *.js",
-    "test": "nyc mocha --require setup-env test/*.test.js",
+    "typings:test": "tsc --noEmit ./index.d.ts",
```
Comment (decompil3d) [inline package.json]:
```suggestion
    "test:types": "tsc --noEmit ./index.d.ts",
```

**Comment by decompil3d** [comment]
@shawnkoon Hi there -- thanks for your contribution. I've tested it out and it looks like it works as expected. I've made some minor suggestions above. Once handled, I can take care of merging/publishing this.

**Comment by shawnkoon** [comment]
Hi @decompil3d, I appreciate your time with reviewing and giving feedbacks 👍 I will take a look at them and make refactors accordingly.

File: package.json (hunk 1 above)
Comment (shawnkoon) [reply package.json]: Great point, I thought the caret in semver would ignore "_minor_._patch_" versions and download the latest `16.x.x` when installed. Do you think it's necessary for us to match those versions?

File: package.json (hunk 2 above)
Comment (shawnkoon) [reply package.json]: Sounds good 👍

**Comment by decompil3d** [comment]
Published as `addhoc@1.2.0`. Thanks for your contribution @shawnkoon!

**Pull request merged**